joblib/numpy_pickle.py

```diff
@@ -231,75 +136,79 @@ def __init__(self, filename, compress=0, cache_size=10, protocol=None):
             np = None
         self.np = np
 
-    def _write_array(self, array, filename):
-        if not self.compress:
-            self.np.save(filename, array)
-            allow_mmap = not array.dtype.hasobject
-            container = NDArrayWrapper(os.path.basename(filename),
-                                       type(array),
-                                       allow_mmap=allow_mmap)
-        else:
-            filename += '.z'
-            # Efficient compressed storage:
-            # The meta data is stored in the container, and the core
-            # numerics in a z-file
-            _, init_args, state = array.__reduce__()
-            # the last entry of 'state' is the data itself
-            with open(filename, 'wb') as zfile:
-                write_zfile(zfile, state[-1], compress=self.compress)
-            state = state[:-1]
-            container = ZNDArrayWrapper(os.path.basename(filename),
-                                            init_args, state)
-        return container, filename
+    def _create_array_wrapper(self, array):
+        """Create and returns a numpy array wrapper from a numpy array.
+
+        Parameters
+        ----------
+        array: numpy.ndarray
+
+        Returns
+        -------
+        wrapper: NumpyArrayWrapper:
+            The numpy array wrapper.
+        """
```

Comment:
Thinking about it, maybe because this is a helper function and the doc is not very informative anyway, just use a single-line for the docstring.